### DISCUSSION TITLE
[typescript-react-apollo] why doesn't it generate mutation hooks? e.g. useCreatePostMutation

### DISCUSSION THREAD

**Discussion by mzalevski** [dotansimha/graphql-code-generator#6266] [Q&A] [answered]
Hi, I was wondering why `typescript-react-apollo` plugin doesn't generate mutation hooks. Am I missing something?

**Comment by ardatan** [comment, marked as answer]
Make sure you have valid operations pointed with `documents`. If you have it already, please share a clear minimal reproduction.

**Reply by mzalevski** [reply]
I am using basic CRA + typescript + apollo. Nothing fancy. Maybe except the fact that I followed hasura docs tutorial on how to configure the codegen [here](https://hasura.io/learn/graphql/typescript-react-apollo/codegen/). Here is my `codegen.js`. I am not sure how can I add `mutations` to `documents` field. I saw the prompt to do so in the docs [here](https://www.graphql-code-generator.com/docs/plugins/typescript-react-apollo) but I was not sure how to comply to it.

```typescript
module.exports = {
  schema: [
    {
      "https://xxx.hasura.app/v1/graphql": {
        headers: {
          "x-hasura-admin-secret":
            "xxx",
        },
      },
    },
  ],
  documents: ["./src/**/*.tsx", "./src/**/*.ts"],
  overwrite: true,
  generates: {
    "./src/generated/graphql.tsx": {
      plugins: [
        "typescript",
        "typescript-operations",
        "typescript-react-apollo",
      ],
      config: {
        skipTypename: false,
        withHooks: true,
        withHOC: false,
        withComponent: false,
      },
    },
    "./graphql.schema.json": {
      plugins: ["introspection"],
    },
  },
};

```

**Reply by ardatan** [reply]
You should have operations in this path `"./src/**/*.tsx", "./src/**/*.ts"`.
https://graphql.org/learn/queries/

**Reply by mzalevski** [reply]
I am introspecting the schema to the `./graphql.schema.json`. `documents: "./graphql.schema.json"` this did not help :/

**Reply by ardatan** [reply]
But ./graphql-schema.json is not an operation file :) Did you check the link? It shows how operations look like?

**Reply by mzalevski** [reply]
oh, ok I got it. I missunderstood you earlier.
`codegen.js` --> `documents: ["./operations.graphql"]`
+
`operations.graphql`
```graphql
mutation InsertPost($object: posts_insert_input!) {
  insert_posts_one(object: $object) {
    id
  }
}
```
worked. I was confused because queries hooks got generated without writing down operations :/

**Reply by ardatan** [reply]
But the naming of that file doesn't seem correct because it is not schema definition but operation definition :)

**Reply by mzalevski** [reply]
true, I will fix this. btw do you have an idea on how to generate such operations from schema? I find myself copying schema parts to construct these :/

**Reply by dotansimha** [reply]
This might give some background and context to your question: https://github.com/dotansimha/graphql-code-generator/discussions/2349
